aws-http-nio-client-netty/src/main/java/software/amazon/awssdk/http/nio/netty/NettyNioAsyncHttpClient.java

```diff
@@ -65,6 +65,9 @@
     NettyNioAsyncHttpClient(NettySdkHttpClientFactory factory, AttributeMap serviceDefaultsMap) {
         this.serviceDefaults = new ServiceDefaults(serviceDefaultsMap);
         this.trustAllCertificates = factory.trustAllCertificates().orElse(Boolean.FALSE);
+        // 0 thread count will use Netty's default number of threads.
+        this.group = new NioEventLoopGroup(factory.eventLoopConfiguration().numberOfThreads().orElse(0),
+                                           factory.eventLoopConfiguration().threadFactory().orElse(null));
```

Comment:
Yeah I was going to tackle that in a separate PR. This just uses Netty's default for now.